python313Packages.{vapoursynth-havsfunc,vsjetpack,jetpytools},vapoursynth-{akarin,zip}: init

Sigmanificient opened the pull request:
## Description of changes

- https://github.com/NixOS/nixpkgs/issues/331513

```
[x] vapoursynth-havsfunc
 └ [x] vsdenoise
    ├ [x] vsscale
    ├ [x] vsmasktools
    │  ├ [x] TCanny
    │  ├ [x] TEdgeMask
    │  ├ [x] AWarpSharp2
    │  └ [x] warpsharp (AWarpSharp2SF)
    └ [x] vsaa
       ├ [x] vskernels
       │  └ [x] vstools
       │     └ [x] stgpytools
       └ [x] vsrgtools
          ├ [x] vspyplugin
          │  └ [x] vsexprtools
          ├ [x] vs-removegrain (RGVS)
          ├ [x] RGSF
          └ [X] Akarin Expr
```

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Sigmanificient committed:
python313Packages.vapoursynth-havsfunc: 33.0.0

Sigmanificient commented:
@ofborg build python312Packages.{stgpytools,vstools,vskernels,vsexprtools,vspyplugin,vsrgtools,vmasktools,vsaa,vsscale,vsdenoise},vs-removegrain,rgsf,akarin-expr,vapoursynth-{tcanny,tedgemask,awapsharp2,vapoursynth-havsfunc},warpshrp

Sigmanificient commented:
Hmm it dont like the way it's written

@ofborg build python312Packages.stgpytools python312Packages.vstools python312Packages.vskernels python312Packages.vsexprtools python312Packages.vspyplugin python312Packages.vsrgtools python312Packages.vmasktools python312Packages.vsaa python312Packages.vsscale python312Packages.vsdenoise vs-removegrain rgsf akarin-expr vapoursynth-tcanny vapoursynth-tedgemask vapoursynth-awapsharp2 vapoursynth-havsfunc warpsharp

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4356

h7x4 commented on pkgs/development/python-modules/vsdenoise/default.nix:
```diff
@@ -0,0 +1,47 @@
+  ];
+
+  meta = {
+    description = "VapourSynth denoising, regression, and motion-compensation functions";
```
```suggestion
    description = "VapourSynth denoising, regression, and motion-compensation functions";
    homepage = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-denoise";
```

h7x4 commented on pkgs/development/python-modules/vsaa/default.nix:
```diff
@@ -0,0 +1,43 @@
+  ];
+
+  meta = {
+    description = "VapourSynth anti aliasing and scaling functions";
```
```suggestion
    description = "VapourSynth anti aliasing and scaling functions";
    homepage = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-aa";
```

h7x4 commented on pkgs/by-name/wa/warpsharp/package.nix:
```diff
@@ -0,0 +1,40 @@
+    description = "fp32 warpsharp for vaporsynth";
+    homepage = "https://github.com/IFeelBloated/warpsharp";
+    maintainers = with lib.maintainers; [ sigmanificient ];
+    # no license, not sure what to put
```
This is yet another one of those with dubious files missing both copyright notice and license. I think it will just have to be marked as `unfree` until further ado.

h7x4 commented on pkgs/by-name/vs/vs-removegrain/package.nix:
```diff
@@ -0,0 +1,44 @@
+    description = "VapourSynth port of RemoveGrain and Repair plugins from Avisynth";
+    homepage = "https://github.com/vapoursynth/vs-removegrain";
+    maintainers = with lib.maintainers; [ sigmanificient ];
+    # no license, not sure what to put
```
This one is weird. `cleanse.cpp` seems to be licensed under MIT, the major part under WTFPL, and `verticalcleaner.cpp` unmarked? Not sure we can assume anything about the last one, might be reasonable to mark as `unfree` until it's cleared up?

h7x4 commented on pkgs/by-name/va/vapoursynth-tedgemask/package.nix:
```diff
@@ -0,0 +1,37 @@
+    description = "Edge detection filter";
+    homepage = "https://github.com/dubhater/vapoursynth-tedgemask";
+    maintainers = with lib.maintainers; [ sigmanificient ];
+    # no license, not sure what to put
```
gpl2Only, see bottom of readme

h7x4 commented on pkgs/by-name/va/vapoursynth-awarpsharp2/package.nix:
```diff
@@ -0,0 +1,35 @@
+    description = "Sharpens by warping";
+    homepage = "https://github.com/dubhater/vapoursynth-awarpsharp2";
+    maintainers = with lib.maintainers; [ sigmanificient ];
+    # no license, not sure what to put
```
ISC, see bottom of readme

h7x4 commented on pkgs/by-name/rg/rgsf/package.nix:
```diff
@@ -0,0 +1,44 @@
+    description = "RGVS Single Precision";
+    homepage = "https://github.com/IFeelBloated/RGSF";
+    maintainers = with lib.maintainers; [ sigmanificient ];
+    # no license, not sure what to put
```
This one also has a few unlicensed files, seemingly from the other dependency? Considering it's probably a fork, I guess `vsremove-grain` needs to be cleared up first? Or is it the other way around?

h7x4 commented on pkgs/by-name/ak/akarin-expr/package.nix:
```diff
@@ -0,0 +1,50 @@
+  '';
+
+  meta = {
+    description = ""; # I have no clue what the description should be :<
```
This probably needs to be addressed. Maybe `An experimental VapourSynth plugin` would do?

Sigmanificient replied:
Yup, I'll jus drop the `An` article

h7x4 commented on pkgs/by-name/vs/vs-removegrain/package.nix:
```diff
@@ -0,0 +1,44 @@
+    substituteInPlace meson.build \
+      --replace-fail \
+        "install_dir : join_paths(vapoursynth_dep.get_pkgconfig_variable('libdir'), 'vapoursynth')," \
+        "install_dir : join_paths('${placeholder "out"}/lib/', 'vapoursynth'),"
```
Wouldn't this work, considering the `'` is already quoted by `"`, and does not have its non-variable-evaluating effect?

```suggestion
        "install_dir : join_paths('$out/lib/', 'vapoursynth'),"
```

Sigmanificient replied:
Both are working!

h7x4 reviewed (commented):
Diff seems mostly clean, but there's a bunch of licensing stuff to figure out. I think we need to mark the ones that are unclear as `unfree` until they are fixed or clarified upstream.

Sigmanificient commented:
Thanks for the returns! Seems like I miss a bunch of hidden licenses :eyes:

snaakey commented:
Wouldn't it make sense to consistently prefix _all_ VapourSynth plugins with `vapoursynth-` ?

emilazy commented:
We should generally use upstream names for packages except when there is ambiguity or it would be really confusing. Categorization is an orthogonal concern to package naming.

Sigmanificient commented:
> Wouldn't it make sense to consistently prefix _all_ VapourSynth plugins with `vapoursynth-` ?

Some plugins on nixpkgs dont have the prefix, so it is a bit unclear, I put the prefix when it was present on the repo name

snaakey commented on pkgs/by-name/vs/vs-removegrain/package.nix:
```diff
@@ -0,0 +1,44 @@
+    description = "VapourSynth port of RemoveGrain and Repair plugins from Avisynth";
+    homepage = "https://github.com/vapoursynth/vs-removegrain";
+    maintainers = with lib.maintainers; [ sigmanificient ];
+    license = lib.license.unfree; # unclear
```
```suggestion
    license = lib.licenses.unfree; # unclear
```

snaakey commented on pkgs/by-name/rg/rgsf/package.nix:
```diff
@@ -0,0 +1,44 @@
+    description = "RGVS Single Precision";
+    homepage = "https://github.com/IFeelBloated/RGSF";
+    maintainers = with lib.maintainers; [ sigmanificient ];
+    license = lib.license.unfree; # unclear license
```
```suggestion
    license = lib.licenses.unfree; # unclear license
```

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/4419

SuperSandro2000 commented on pkgs/development/python-modules/stgpytools/default.nix:
```diff
@@ -0,0 +1,40 @@
+  buildPythonPackage,
+  fetchFromGitHub,
+  setuptools,
+
```
```suggestion
```

SuperSandro2000 commented on pkgs/by-name/vs/vs-removegrain/package.nix:
```diff
@@ -0,0 +1,44 @@
+
+  buildInputs = [ vapoursynth ];
+  nativeBuildInputs = [
+    meson
+    cmake
+    pkg-config
+    ninja # resolves: Could not detect Ninja
```
Generally we only need 3 of those, sometimes only 2.

SuperSandro2000 commented on pkgs/by-name/vs/vs-removegrain/package.nix:
```diff
@@ -0,0 +1,44 @@
+    description = "VapourSynth port of RemoveGrain and Repair plugins from Avisynth";
+    homepage = "https://github.com/vapoursynth/vs-removegrain";
+    maintainers = with lib.maintainers; [ sigmanificient ];
+    license = lib.licenses.unfree; # unclear
```
Maybe wtfpl2 https://github.com/vapoursynth/vs-removegrain/blob/89ca38a6971e371bdce2778291393258daa5f03b/src/shared.cpp#L12 ?

SuperSandro2000 commented on pkgs/development/python-modules/vsaa/default.nix:
please rebase and fixup this

SuperSandro2000 commented on pkgs/by-name/rg/rgsf/package.nix:
```diff
@@ -0,0 +1,44 @@
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  name = "rgsf";
```
```suggestion
  pname = "rgsf";
```

SuperSandro2000 commented on pkgs/by-name/ak/akarin-expr/package.nix:
```diff
@@ -0,0 +1,50 @@
+}:
+
+stdenv.mkDerivation {
+  name = "akarin-expr";
```
Please use pname.

SuperSandro2000 commented on pkgs/by-name/ak/akarin-expr/package.nix:
```diff
@@ -0,0 +1,50 @@
+    hash = "sha256-MlQ0iOYPMAGttOVD2yF4kyZE+cpQ240DC2TBJY8/pH4=";
+  };
+
+  nativeBuildInputs = [
+    meson
+    pkg-config
+    cmake
+    ninja
+  ];
```
Please check what is really needed here

SuperSandro2000 commented on pkgs/development/python-modules/vsrgtools/default.nix:
This should also be done in the initial commit

SuperSandro2000 commented on pkgs/by-name/va/vapoursynth-tcanny/package.nix:
```diff
@@ -0,0 +1,46 @@
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  name = "vapoursynth-tcanny";
```
pname

SuperSandro2000 commented on pkgs/by-name/va/vapoursynth-tedgemask/package.nix:
```diff
@@ -0,0 +1,37 @@
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  name = "vapoursynth-tedgemask";
```
pname

SuperSandro2000 commented on pkgs/development/python-modules/vsmasktools/default.nix:
also fixup

SuperSandro2000 reviewed (commented):
Please double check that everything has pythonImportsCheck.

Also please cleanup the commit messages. When squashing you seem to have left some mesages.

nyabinary commented:
There hasn't been a release for 4 years for havsfunc, would it be worth it atp to just use the latest gitc ommit?

Sigmanificient commented:
@nyabinary Asked the maintainer

FliegendeWurst commented on pkgs/development/python-modules/vstools/default.nix:
```diff
@@ -0,0 +1,49 @@
+
+buildPythonPackage rec {
+  pname = "vstools";
+  version = "3.0.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Jaded-Encoding-Thaumaturgy";
+    repo = "vs-tools";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-XqPIh2iTxcIU3o8XIrsE1o4B8D/YkP7nlJB40woG+so=";
+  };
```
```suggestion
  version = "3.3.3";
  pyproject = true;

  src = fetchFromGitHub {
    owner = "Jaded-Encoding-Thaumaturgy";
    repo = "vs-tools";
    rev = "refs/tags/v${version}";
    hash = "sha256-YqRUBGCJcEXeYzlHq4zSEpVAXZhuxV9xA1Exfs6RLAI=";
  };
```

FliegendeWurst commented on pkgs/by-name/ak/akarin-expr/package.nix:
```diff
@@ -0,0 +1,49 @@
+  nativeBuildInputs = [
+    meson
+    pkg-config
+    ninja
```
```suggestion
    ninja
    llvm_18
```

FliegendeWurst commented on pkgs/development/python-modules/stgpytools/default.nix:
```diff
@@ -0,0 +1,39 @@
+
+buildPythonPackage rec {
+  pname = "stgpytools";
+  version = "1.0.0";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Setsugennoao";
+    repo = "stgpytools";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-YxDQmtyJgYDUcvMxQ4KqA4yYLO8N4fzCyayyBnuZjf0=";
+  };
```
```suggestion
  version = "1.2.2";
  pyproject = true;

  src = fetchFromGitHub {
    owner = "Setsugennoao";
    repo = "stgpytools";
    rev = "1cfcf8b26315db706ad1947b4ab3ef938a857061"; # upstream did not tag release
    hash = "sha256-tJTq4xPxuFAZLzhworjjixFNWDb0FfBKLBLnttFYvH4=";
  };
```

FliegendeWurst commented on pkgs/by-name/va/vapoursynth-tcanny/package.nix:
```diff
@@ -0,0 +1,46 @@
+  ];
+
+  meta = {
+    description = "Filter for VapourSynth";
```
```suggestion
    description = "Canny edge detection for VapourSynth";
```

Pandapip1 commented on pkgs/development/python-modules/vsmasktools/default.nix:
```diff
@@ -0,0 +1,52 @@
+
+  meta = {
+    description = "Various masking tools for VapourSynth";
+    homepage = "https://vsmasktools.encode.moe";
```
This fails domain resolution for me.

Pandapip1 replied:
```suggestion
    homepage = "https://vsmasktools.encode.moe";
    changelog = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-masktools/releases";
```

Pandapip1 commented on pkgs/development/python-modules/vspyplugin/default.nix:
```diff
@@ -0,0 +1,42 @@
+
+  meta = {
+    description = "RgTools and related functions";
+    homepage = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-pyplugin";
```
```suggestion
    homepage = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-pyplugin";
    changelog = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-pyplugin/releases";
```

Pandapip1 commented on pkgs/development/python-modules/vsrgtools/default.nix:
```diff
@@ -0,0 +1,55 @@
+
+  meta = {
+    description = "RgTools and related functions";
+    homepage = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-rgtools";
```
```suggestion
    homepage = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-rgtools";
    changelog = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-rgtools/releases";
```

Pandapip1 commented on pkgs/development/python-modules/vsscale/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+  meta = {
+    description = "VapourSynth (de)scale functions";
+    homepage = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-scale";
```
```suggestion
    homepage = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-scale";
    changelog = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-scale/releases";
```

Pandapip1 commented on pkgs/development/python-modules/vstools/default.nix:
```diff
@@ -0,0 +1,49 @@
+
+  meta = {
+    description = "Functions and utils related to VapourSynth";
+    homepage = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-tools";
```
```suggestion
    homepage = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-tools";
    changelog = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-tools/releases";
```

Pandapip1 commented on pkgs/by-name/rg/rgsf/package.nix:
```diff
@@ -0,0 +1,44 @@
+
+    runHook postInstall
+  '';
+
```
```suggestion

    passthru.updateScript = nix-update-script { };

```

Requires adding `nix-update-script` to package inputs.

Pandapip1 commented on pkgs/by-name/va/vapoursynth-awarpsharp2/package.nix:
```diff
@@ -0,0 +1,35 @@
+    pkg-config
+    ninja # resolves: Could not detect Ninja
+  ];
+
```
```suggestion

    passthru.updateScript = nix-update-script { };

```

Requires adding `nix-update-script` to package inputs.

Pandapip1 commented on pkgs/by-name/va/vapoursynth-tcanny/package.nix:
```diff
@@ -0,0 +1,46 @@
+    pkg-config
+    ninja # resolves: Could not detect Ninja
+  ];
+
```
```suggestion

    passthru.updateScript = nix-update-script { };

```

Requires adding `nix-update-script` to package inputs.

Pandapip1 commented on pkgs/by-name/va/vapoursynth-tedgemask/package.nix:
```diff
@@ -0,0 +1,37 @@
+    pkg-config
+    ninja # resolves: Could not detect Ninja
+  ];
+
```
```suggestion

    passthru.updateScript = nix-update-script { };

```

Requires adding `nix-update-script` to package inputs.

Pandapip1 commented on pkgs/development/python-modules/vstools/default.nix:
```diff
@@ -0,0 +1,49 @@
+  ];
+
+  doCheck = false; # no tests
+
```
```suggestion

    passthru.updateScript = nix-update-script { };

```

Requires adding `nix-update-script` to package inputs.

FliegendeWurst commented on pkgs/development/python-modules/vsmasktools/default.nix:
```diff
@@ -0,0 +1,52 @@
+    vsexprtools
+    vsrgtools
+  ];
+
```
All Python modules already have a pypi-based update script, I think.

TRPB commented:
what needs to happen to get this merged? I'd like to use this package natively if possible

FliegendeWurst commented:
> what needs to happen to get this merged? I'd like to use this package natively if possible

The review comments need to be addressed. And I think it would be easier to submit one new Python package per PR.

If you're interested in this package, you can clone their branch and submit as another PR (after applying most of the review suggestions).

Sigmanificient commented on pkgs/development/python-modules/stgpytools/default.nix:
```diff
@@ -0,0 +1,39 @@
+    "stgpytools.utils"
+  ];
+
+  doCheck = false; # no tests
```
No, it doesn't ! Although `pythonImportsCheck` is not present which should be fixed indeed

Sigmanificient committed:
vapoursynth: revert version bump (fix python binding package)

Sigmanificient committed:
python313Packages.jetpytools: init at 1.4.0

Sigmanificient committed:
vapoursynth-akarin: init at 0.95

Sigmanificient committed:
vapoursynth-zip: init at 7

Sigmanificient committed:
python313Packages.vsjetpack: init at 0.4.0

Sigmanificient commented:
Hi, to be honest the amount of change requested discouraged me to work on the pr (at a time where i was less comfortable with nix). I also had not realized that is clearly a lot of duplication in them :sweat_smile: 

Anyway, the pr dep tree has changed quite a bunch in the meantime, so I updated it too:
```
[x] vapoursynth-havsfunc
 └ [x] vsdenoise
    ├ [x] vsscale
    ├ [x] vsmasktools
    │  ├ [x] TCanny
    │  ├ [x] TEdgeMask
    │  ├ [x] AWarpSharp2
    │  └ [x] warpsharp (AWarpSharp2SF)
    └ [x] vsaa
       ├ [x] vskernels
       │  └ [x] vstools
       │     └ [x] stgpytools
       └ [x] vsrgtools
          ├ [x] vspyplugin
          │  └ [x] vsexprtools
          ├ [x] vs-removegrain (RGVS)
          ├ [x] RGSF
          └ [X] Akarin Expr
```
now is:
```
[x] vapoursynth-havsfunc
 └ [x] vsjetpack
    ├ [x] vapoursynth-akarin ("Akarin Expr" on the previous tree)
    └ [x] vapoursynth-zip

[x] replaced by vsjetpack
 ├ [x] vsdenoise
 ├ [x] vsscale
 ├ [x] vsaa
 ├ [x] vsmasktools
 ├ [x] vsrgtools
 ├ [x] vsexprtools
 ├ [x] vskernels
 └ [x] vstools

[ ] Old deps
├ [ ] vsmasktools deps
│  └ [ ] TCanny
│     ├ [ ] TEdgeMask 
│     ├ [ ] AWarpSharp2
│     └ [ ] warpsharp (AWarpSharp2SF)
├ [ ] vstools deps
│  └ [ ] stgpytools
└ [ ] vsrgtools
   ├ [ ] vspyplugin
   ├ [ ] vs-removegrain (RGVS)
   ├ [ ] RGSF
   └ [ ] Akarin Expr
```

Sigmanificient commented on pkgs/by-name/va/vapoursynth/package.nix:
```diff
@@ -17,13 +17,13 @@
 stdenv.mkDerivation rec {
   pname = "vapoursynth";
-  version = "72";
+  version = "71";
```
> [!NOTE]
> This change is done because the `vapoursynth` update broke `python3Packages.vapoursynth`
- see: https://github.com/NixOS/nixpkgs/pull/413321

Pandapip1 commented on pkgs/development/python-modules/jetpytools/default.nix:
```diff
@@ -0,0 +1,41 @@
+
+  meta = {
+    description = "RgTools and related functions";
+    homepage = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-pyplugin";
```
```suggestion
    homepage = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-pyplugin";
    changelog = "https://github.com/Jaded-Encoding-Thaumaturgy/vs-pyplugin/releases";
```

Pandapip1 commented on pkgs/by-name/va/vapoursynth-zip/package.nix:
```diff
@@ -0,0 +1,40 @@
+    description = "VapourSynth Zig Image Process";
+    homepage = "https://github.com/dnjulek/vapoursynth-zip";
+    license = lib.licenses.mit;
+    maintainers = [ ];
```
Why aren't you putting yourself as a maintainer here?

Pandapip1 commented on pkgs/by-name/va/vapoursynth-zip/package.nix:
```diff
@@ -0,0 +1,40 @@
+
+  meta = {
+    description = "VapourSynth Zig Image Process";
+    homepage = "https://github.com/dnjulek/vapoursynth-zip";
```
```suggestion
    homepage = "https://github.com/dnjulek/vapoursynth-zip";
    changelog = "https://github.com/dnjulek/vapoursynth-zip/releases";
```

Pandapip1 commented on pkgs/by-name/va/vapoursynth-akarin/package.nix:
```diff
@@ -0,0 +1,48 @@
+  ];
+
+  meta = {
+    description = "Enhanced LLVM-based std.Expr plugin for vapoursynth";
```
Missing homepage and platforms.
```suggestion
    description = "Enhanced LLVM-based std.Expr plugin for vapoursynth";
    homepage = "https://github.com/AkarinVS/vapoursynth-plugin";
    changelog = "https://github.com/AkarinVS/vapoursynth-plugin/releases";
    platforms = lib.platforms.all; # I assume?
```

Pandapip1 commented on pkgs/by-name/va/vapoursynth-zip/package.nix:
```diff
@@ -0,0 +1,40 @@
+  meta = {
+    description = "VapourSynth Zig Image Process";
+    homepage = "https://github.com/dnjulek/vapoursynth-zip";
+    license = lib.licenses.mit;
```
Missing platforms
```suggestion
    license = lib.licenses.mit;
    platforms = lib.platforms.all;
```

FliegendeWurst commented on pkgs/by-name/va/vapoursynth-zip/package.nix:
```diff
@@ -0,0 +1,42 @@
+    hash = "sha256-/bHD+t5upQ+So1k+5/IbNTfeugMqcnlDlaH/H9nXgR8=";
+  };
+
+  nativeBuildInputs = [
+    zig.hook
+  ];
```
```suggestion
  buildInputs = [
    zig.hook
  ];
```
Required for `strictDeps = true` support (#178468)
